examples/TPMS/grading/cell_type.py

```diff
@@ -1,58 +1,77 @@
+"""Example of morphing between multiple TPMS surface types."""
+
+from typing import Callable, List, Tuple
+
 import numpy as np
+import pyvista as pv
 
 from microgen import Tpms
-from microgen.shape.surface_functions import fischerKochS, gyroid, schwarzP
+from microgen.shape.surface_functions import fischerKochS, gyroid, schwarzD, schwarzP
 
-repeat = 2
-k = 2.0
-point_1 = (-0.5, 0, 0)
-point_2 = (0.5, 0, 0)
-point_3 = (0, 0.5, 0)
+REPEAT = (4, 4, 2)
+POINTS = [(-0.5, 0, 0), (0.5, 0, 0), (0, 0.5, 0), (0, -0.5, 0)]
+SURFACES = [schwarzP, gyroid, fischerKochS, schwarzD]
 
 
-def exp_func(x, y, z):
+def exp_func(x: np.ndarray, y: np.ndarray, z: np.ndarray) -> np.ndarray:
+    """Transition between points using exponential function."""
+    k = 0.5
     norm = x**2 + y**2 + z**2
     return 1 + np.exp(k * norm)
 
 
-def weight(x, y, z, points, index):
-    denom = 0.0
+def weight(
+    x: np.ndarray,
+    y: np.ndarray,
+    z: np.ndarray,
+    points: List[Tuple[float, float, float]],
+    index: int,
+) -> np.ndarray:
+    """Weight function for morphing between points."""
+    denom = np.zeros_like(x)
     for p in points:
```

Comment:
There is a way to replace this for loop using numpy operations ?

Reply:
`points` is renamed to `control_points`. The number of points is low and points are tuples so I think it can remains like this.